app/models/host/discovered.rb

```diff
@@ -55,6 +55,18 @@ def self.import_host_and_facts facts
     # and save (interfaces are created via puppet parser extension)
     host.save(:validate => false) if host.new_record?
     state = host.import_facts(facts)
+
+    # lock the host into discovery via PXE, if feature is enabled in settings
+    if Foreman::Cast.to_bool(Setting['discovery_lock']) and !Setting['discovery_lock_template'].blank?
```

Comment:
@imriz I see the discussion on the issue suggested a common snippet to use in both templates, maybe the `discovery_lock_template` should have a default with that snippet?